gcloud-java-datastore/src/test/java/com/google/gcloud/datastore/LocalGcdHelper.java

```diff
@@ -115,6 +212,84 @@ public static ProcessStreamReader start(Process process, String blockUntil) thro
       return thread;
     }
   }
+  
+  private static class CommandWrapper {
+    
+    private final List<String> prefix;
+    private List<String> command;
+    private String nullFilename;
+    private boolean redirectOutputToNull;
+    private boolean redirectErrorStream;
+    private boolean redirectErrorInherit;
+    private Path directory;
+    
+    private CommandWrapper() {
+      this.prefix = new ArrayList<>();
+    }
+    
+    public CommandWrapper command(String... command) {
+      this.command = new ArrayList<>(command.length + this.prefix.size());
+      this.command.addAll(prefix);
+      this.command.addAll(Arrays.asList(command));
+      return this;
+    }
+    
+    public CommandWrapper redirectOutputToNull() {
+      this.redirectOutputToNull = true;
+      return this;
+    }
+    
+    public CommandWrapper redirectErrorStream() {
+      this.redirectErrorStream = true;
+      return this;
+    }
+    
+    public CommandWrapper redirectErrorInherit() {
+      this.redirectErrorInherit = true;
+      return this;
+    }
+    
+    public CommandWrapper directory(Path directory) {
+      this.directory = directory;
+      return this;
+    }
+    
+    public ProcessBuilder builder() {
+      ProcessBuilder builder = new ProcessBuilder(command);
+      if (redirectOutputToNull) {
+        builder.redirectOutput(new File(nullFilename));
+      }
+      if (directory != null) {
+        builder.directory(directory.toFile());
+      }
+      if (redirectErrorStream) {
+        builder.redirectErrorStream(true);
+      }
+      if (redirectErrorInherit) {
+        builder.redirectError(ProcessBuilder.Redirect.INHERIT);
+      }
+      return builder;
+    }
+    
+    public Process start() throws IOException {
+      return builder().start();
+    }
+    
+    public static CommandWrapper cmd() {
+      CommandWrapper wrapper = new CommandWrapper();
```

Comment:
Would it be cleaner to have the constructor call isWindows() and initialize prefix and nullFilename?
bash() and cmd() need to make sure they initialize things correctly (e.g. nullFilename) to avoid NPE in builder() line 260.

I know that it means that in one case there are basically twice the isWindows() checks. once for picking the script name and one here but I think it would simplify the code of the callers. just use contractor or single factory method.

If you don't like that then maybe at least change redirectOutputToNull to redirectOutput(String or File) and builder() method redirect if not null.
check if string is